ml-proto/README.md

```diff
@@ -187,22 +187,22 @@ case:
   ( case <value> <expr>* fallthrough? )  ;; = (case <int> (block <expr>*) fallthrough?)
   ( case <value> )                       ;; = (case <int> (nop) fallthrough)
 
-func:   ( func <name>? <param>* <result>? <local>* <expr>* )
-param:  ( param <type>* ) | ( param <name> <type> )
-result: ( result <type> )
-local:  ( local <type>* ) | ( local <name> <type> )
-
-module: ( module <func>* <global>* <export>* <table>* <memory>? <data>* )
-export: ( export "<char>*" <var> )
-global: ( global <type>* ) | ( global <name> <type> )
-table:  ( table <var>* )
-memory: ( memory <int> <int>? )
-data:   ( data "<char>*" )
+func:    ( func <name>? <param>* <result>? <local>* <expr>* )
+param:   ( param <type>* ) | ( param <name> <type> )
+result:  ( result <type> )
+local:   ( local <type>* ) | ( local <name> <type> )
+
+module:  ( module <func>* <global>* <export>* <table>* <memory>? )
+export:  ( export "<char>*" <var> )
+global:  ( global <type>* ) | ( global <name> <type> )
+table:   ( table <var>* )
+memory:  ( memory <int> <int>? <segment>* )
+segment: ( segment <int> "<char>*" )
 ```
 
 Here, productions marked with respective comments are abbreviation forms for equivalent expansions.
 
-The data string is used to initialise the lower end of the memory. It is an ASCII string, that can have the usual escape sequences, or hex escapes of the form `\xx` to denote a single byte.
+The segment string is used to initialise the lower end of the memory. It is an ASCII string, that can have the usual escape sequences, or hex escapes of the form `\xx` to denote a single byte.
```

Comment:
What's the "usual" escape sequences? I'd like to have something a bit more precise!
Is there a pre-existing standard we can use? Presumably Unicode would work.
How are quotes escaped?

I guess we don't define `char` at all here, which is the true problem.